### PR TITLE
`azurerm_api_management` - Restore request does not need to handle the response which does not have

### PR DESCRIPTION
**Pull request by xuzhang3**
Fixes: #17206 
This recover request is different from the create/update. The response does not have any response body or HTTP headers like create/update. A successful recover request will have an HTTP status code of 201. Handle the response error is enough. 

```log
=== RUN   TestAccApiManagement_softDeleteRecovery
=== PAUSE TestAccApiManagement_softDeleteRecovery
=== CONT  TestAccApiManagement_softDeleteRecovery
--- PASS: TestAccApiManagement_softDeleteRecovery (959.33s)
PASS

```

### CODE REVIEW

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @xuzhang3, this looks like an api bug if the future isn't working correctly. I believe we'll have to merge this the way it is since it is broken but can you open an api ticket and link that into a comment on this PR so we can track when it'll be fixed?

**Comment by xuzhang3** [comment]
APIM issue: https://github.com/Azure/azure-rest-api-specs/issues/23456

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @xuzhang3 in future please make sure to leave a comment in the code linking to the raised issue. LGTM 👍

**Review by stephybun — APPROVED** [review]

File: internal/services/apimanagement/api_management_resource.go [hunk 1]
```diff
@@ -729,13 +729,11 @@ func resourceApiManagementServiceCreateUpdate(d *pluginsdk.ResourceData, meta in
 				},
 			}
 
-			future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ServiceName, params)
+			// We only handle the error here because no request body is returned https://github.com/Azure/azure-rest-api-specs/issues/23456
+			_, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ServiceName, params)
```
Comment (tombuildsstuff) [inline internal/services/apimanagement/api_management_resource.go:733]: To give a little context here: this is returning an LRO, so there won't be a HTTP Response **by design** - this API behaviour is intentional and not an API bug. If you wish to query the state of a resource that's being provisioned as an LRO, you'll need to either poll on the LRO (using `WaitForCompletion` in the Track1 SDK) - or by polling on the resource and checking the `provisioningState` (as below) - although the LRO shouldn't be returning until this is completed, therefore polling on the LRO **should** be sufficient here

**Pull request merged**